File: c/src/common/bostonfishchowder.c
#include "bostonfishchowder.h"

soup_t * bostonfishchowder_new()
{
	soup_t * soup = soup_new();
	soup->name = "ScrodFishChowder";
	arraylist_string_add( soup->ingredients,"1 Pound Fresh Scrod");
	arraylist_string_add( soup->ingredients,"1 cup corn");    
	arraylist_string_add( soup->ingredients,"1/2 cup heavy cream");
	arraylist_string_add( soup->ingredients,"1/4 cup butter");    
	arraylist_string_add( soup->ingredients,"1/4 cup potato chips");
	return soup;
}

File: clojure/src/common/bostonfishchowder.clj
(defclass bostonfishchowder
   (fishchowder)
   ())

(defmethod initialize-instance :after ((this bostonfishchowder) &key)
   (setf (slot-value this 'name) "ScrodFishChowder")
   (setf (slot-value this 'ingredients) (list
       "1 Pound Fresh Scrod"
       "1 cup corn"
       "1/2 cup heavy cream"
       "1/4 cup butter"
       "1/4 cup potato chips"
	)))
